### PR TITLE
feat: Create dash and ham converters.

### PR DESCRIPTION
**Pull request by DuckyCB**
## Description
- Create DashManifest types
- Lint classes
- Create converter from mpd to ham
- Create converter from ham to mpd

## Requirements Checklist
- [ ] Unit Tests updated or fixed
- [ ] Docs/guides updated

### CODE REVIEW

File: lib/src/cmaf/utils/dash/mpd.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	return parseString(raw, (err: Error | null, result: DashManifest) => {
+		if (err) {
+			throw new Error(err.message);
+		}
+		replace(result);
+	});
+}
+
+// export function parseMpdVideo(text: string, uri: string) {
+// 	const parsedDash = parse(text, { uri });
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/utils/dash/mpd.ts]: Is this necessary to leave this commented?

File: package.json [hunk 2]
```diff
@@ -60,7 +60,9 @@
 		"typescript": "5.2.2"
 	},
 	"dependencies": {
+		"@types/xml2js": "^0.4.14",
 		"m3u8-parser": "^7.1.0",
-		"mpd-parser": "^1.3.0"
+		"mpd-parser": "^1.3.0",
```
Comment (NoeliaBentancor) [inline package.json]: Are we using mpd-parser dependency now?

File: lib/src/cmaf/ham/manifestParser.ts [hunk 3]
```diff
@@ -1,14 +1,29 @@
-async function readHLS(manifestUrl: string): Promise<string> {
-    const response = await fetch(manifestUrl, {
-        headers: {
-            'Content-Type': 'application/vnd.apple.mpegurl',
-        }
-    });
-    return response.text();
+import { parseMpd } from '../utils/dash/mpd';
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/manifestParser.ts]: When this is compiled, reference does not work. Should be .js file.

File: lib/src/cmaf/ham/manifestParser.ts [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
 
 
 export async function m3u8toHam(url: string) {
-    
+
 }
 
+
+export async function mpdToHam(manifest: string) {
+	let dashManifest: DashManifest | undefined;
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/manifestParser.ts:21]: We should have a function for reading dash manifests and then converting into ham object

File: lib/src/cmaf/utils/dash/DashManifest.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		subsegmentStartsWithSAP?: string,
+		audioSamplingRate?: number,
+		mimeType?: string,
+		codecs: string,
+		startWithSAP?: string,
+		par?: string;
+		sar?: string;
+		maxWidth?: number;
+		maxHeight?: number;
+
```
Comment (mpollingerQualabs) [inline lib/src/cmaf/utils/dash/DashManifest.ts:44]:
```suggestion
```

File: lib/src/cmaf/ham/hlsManifest.ts
Comment (mpollingerQualabs) [inline lib/src/cmaf/ham/hlsManifest.ts:1]: Might not want to apply these changes.
We need to take a stab at the linter for the whole project.

File: lib/src/cmaf/utils/dash/DashManifest.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		}
+	]
+}
+
+type Representation = {
+	$: {
+		id: string,
+		bandwidth: number,
+	},
+	BaseURL?: string[],
```
Comment (mpollingerQualabs) [inline lib/src/cmaf/utils/dash/DashManifest.ts:21]: Is the `BaseURL` an actual array? Or is it this way to consider content-steering in the future?
I wonder if it shouldn't be just a string, but if it has content-steering then we would have an array of `BaseURL`s
Might need to think about this a bit more

**Review by mpollingerQualabs — COMMENTED** [review]
We should probably take a look at the linter for the project. Also, change the spacing from 8 to 4 (better to use tab actually).

**Review by matiasrb97 — APPROVED** [review]

File: lib/src/cmaf/ham/manifestParser.ts (hunk 4 above)
Comment (matiasrb97) [reply lib/src/cmaf/ham/manifestParser.ts:21]: Let's create a new ticket for the new changes needed here. @DuckyCB

**Pull request merged**